sql/src/main/java/io/crate/executor/transport/ExecutionNodesTask.java

```diff
@@ -273,14 +273,20 @@ private void addCloseContextCallback(TransportCloseContextNodeAction transportCl
             return;
         }
         final ContextCloser contextCloser = new ContextCloser(transportCloseContextNodeAction);
-        result.addListener(new Runnable() {
+        Futures.addCallback(Futures.allAsList(results), new FutureCallback<List<TaskResult>>() {
```

Comment:
I'll need results here anyway with the 1 job init per plan feature, thats why I told bogi to do it like that